httpclient/metrics_round_tripper.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package httpclient
+
+import (
+	"fmt"
+	"github.com/acronis/go-appkit/httpserver/middleware"
+	"github.com/acronis/go-appkit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+var (
+	ExternalHTTPRequestDuration *prometheus.HistogramVec
+	ClassifyRequest             func(r *http.Request, reqType string, logger log.FieldLogger) string
+)
+
+// MustInitAndRegisterMetrics initializes and registers external HTTP request duration metric.
+// Panic will be raised in case of error.
+func MustInitAndRegisterMetrics(namespace string) {
+	ExternalHTTPRequestDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: namespace,
+			Name:      "external_http_request_duration",
```

Comment:
The name external_http_request_duration could be ambiguous, as “external” might imply either incoming requests from the internet or outgoing requests. I suggest renaming the Prometheus metric to http_client_request_duration_seconds for better clarity.

Additionally, this aligns with the naming convention already used in [go-authkit](https://github.com/acronis/go-authkit/blob/main/internal/metrics/metrics.go#L111C18-L111C54).